# Diabetes Detection submission by Prem Bharwani

---


### Importing all the modules


In [29]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn import metrics

### Load the csv file

In [30]:
#note that I have locally stored the dataset file in the same directory as the notebook
Data = pd.read_csv("diabetes2.csv")
print(Data.isna().any().any())
print(Data.isnull().any().any())

False
False


### Please note that the dataset has NO missing values ,No preprocessing required so we're good to go !

### Seperating the features and the target

In [31]:
#Getting the "Outcome" column and then Dropping the column 'Outcome' from the Data (Axis=1 indicates that the column should be dropped!)
y=Data["Outcome"]
X=Data.drop("Outcome",axis=1)
# X.head()
# type(y)

### Splitting in the given ratio using the train_test_split function of sklearn

In [32]:
x_train , x_test , y_train , y_test = train_test_split( X, y , test_size=0.2 , random_state=4)
# len(x_train)
# y_train.head()

### Training the model using Logistic Regression

In [33]:
model = LogisticRegression(max_iter=500)
model.fit(x_train,y_train)

LogisticRegression(max_iter=500)

In [34]:
predictions = model.predict(x_test)
model_score = model.score(x_test,y_test)
print(model_score)
# help(model)

0.8116883116883117


## Accuracy on Test datasets

In [35]:
arr1=predictions.tolist()
arr2=(y_test).tolist()
correct=0
for x in range(len(arr1)):
    if arr1[x]==arr2[x]:
        correct+=1
op_str="Accuracy on the test dataset : {}/{} ({:.2%})".format(correct,len(arr1),model_score)
op_str

'Accuracy on the test dataset : 125/154 (81.17%)'

## Okay so we got a score of 125/154 , which amounts to ~81.17% accuracy !